src/map/chat.c

```diff
@@ -255,6 +255,14 @@ static int chat_leavechat(struct map_session_data *sd, bool kicked)
 	}
 
 	if( leavechar == 0 && cd->owner->type == BL_PC ) {
+
+		// check if new location are CELL_CHKNOCHAT
+		if (map->getcell(cd->usersd[0]->bl.m, NULL, cd->usersd[0]->bl.x, cd->usersd[0]->bl.y, CELL_CHKNOCHAT)) {
+			for (i = (cd->users - 1); i >= 0; i--)
```

Comment:
Is there a reason the loop have to be descending instead of ascending?

Reply:
doesn't have to shift the array.
just my own preference when delete data from array.

previously i tried with `chat->npckickall(...)` but no idea, it crashed for unknown reason.
btw...  `chat->npckickall(...)` also did go through the loop descending when kicking member, maybe the same reason as stated above.

Reply:
Ah, this makes perfect sense then.